tests/Microsoft.Identity.Test.Unit/AuthorityAliasesTests.cs

```diff
@@ -182,5 +182,35 @@ private async Task ValidateCacheEntitiesEnvironmentAsync(ITokenCacheInternal cac
                 Assert.AreEqual(expectedEnvironment, new Uri(kvp.Key.Authority).Host);
             }
         }
+
+        [TestMethod]
+        public void AuthorityNotIncludedInAliasesTestAsync()
+        {
+            //Make sure MSAL is able to create an entry for instance discovery when the configured environment is not present in the 
+            //instance discovery metadata. This is for non-public cloud scenarios. See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2701
+
+            using (var harness = base.CreateTestHarness())
+            {
+                PublicClientApplication app = PublicClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                                            .WithAuthority(new Uri(TestConstants.AuthorityCommonPpeAuthority), true)
+                                                                            .WithHttpManager(harness.HttpManager)
+                                                                            .WithTelemetry(new TraceTelemetryConfig())
+                                                                            .BuildConcrete();
+                app.ServiceBundle.ConfigureMockWebUI();
+
+                //Adding one instance discovery response to ensure the cache is hit for the subsiquent requests.
+                //If MSAL tries to do an additional request this test will fail.
+                harness.HttpManager.AddInstanceDiscoveryMockHandler();
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(TestConstants.AuthorityCommonPpeAuthority);
+
+                AuthenticationResult result = app
+                    .AcquireTokenInteractive(TestConstants.s_scope)
+                    .ExecuteAsync(CancellationToken.None)
+                    .Result;
+
+                Assert.IsNotNull(result);
+                Assert.IsTrue(harness.HttpManager.QueueSize == 0);
+            }
```

Comment:
nit: `Assert.AreEqual(0, httpManager.QueueSize);` looks like most test are doing this.

Reply:
It's not needed at all. The `using` statement at line 192 ensures that once the block of code finishes (at line 213), Dispose() gets called on the harness, which calls Dispose on the HttpManager, which Asserts that the queue is empty.

See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/0d53f3e80e29f9c6d172e069c10e02d71e24f65e/tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpManager.cs#L48

Reply:
that's what i thought, that we processed it automatically somewhere, but i couldn't find it. those other tests should probably be updated at some point.